searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -584,6 +655,10 @@ createForDirectWSetImpl(const IAttributeVector * attribute, V && vector, vespali
     if (!attribute->isImported() && (iattr != nullptr) && supportsGetRawValues<A, VT>(*iattr)) {
         auto * exactA = dynamic_cast<const ExactA *>(iattr);
         if (exactA != nullptr) {
+            if (extractSize(vector) == 1) {
+                const auto & elem = extractElem(vector, 0ul);
```

Comment:
could let auto resolve to a value here and move it into the constructor

Reply:
Just removed the temporary.